### PR TITLE
Track profile switching logic + Glonass support

### PR DESCRIPTION
**Pull request by adel-mamin**
This work is done under the issue https://github.com/swift-nav/exafore_planning/issues/299
Includes https://github.com/swift-nav/peregrine/pull/54, https://github.com/swift-nav/peregrine/pull/53, https://github.com/swift-nav/peregrine/pull/51, https://github.com/swift-nav/peregrine/pull/50, https://github.com/swift-nav/peregrine/pull/47
Requires swift-nav/libswiftnav#365

### CODE REVIEW

**Comment by adel-mamin** [comment]
@valeri-atamaniouk

File: peregrine/defaults.py [hunk 1]
```diff
@@ -253,6 +1264,19 @@
 # disable lock detect
 l1ca_lock_detect_params_disable = {"k1": 0.02, "k2": 1e-6, "lp": 1, "lo": 1}
 
+lock_detect_params_slow = {"k1": 0.005, "k2": 1.4, "lp": 200, "lo": 50}
```
Comment (valeri-atamaniouk) [inline peregrine/defaults.py]: Normally lo >> lp. I.e. lp: 240, lo: 50.

**Comment by valeri-atamaniouk** [comment]
lgfm. If TS gets merged before this, it should be preferred over existing Py equivalents.

File: peregrine/defaults.py (hunk 1 above)
Comment (adel-mamin) [inline peregrine/defaults.py]: Should be fixed now.

**Pull request closed without merging**